feat: Update Releases docs with new sentry-cli set-commits behaviour (#1813)
getsentry/sentry-docs: src/collections/_documentation/workflow/releases/index.md

If you don't want Sentry to connect to your repository, or you're using an unsupported repository provider or VCS (e.g. Perforce), you can alternatively tell Sentry about your raw commit metadata via the API using the [create release endpoint](/api/releases/post-organization-releases/).

#### Using the CLI
```bash
# Assumes you're in a git repository
export SENTRY_AUTH_TOKEN=...
export SENTRY_ORG=my-org
VERSION=$(sentry-cli releases propose-version)

# Create a release
sentry-cli releases new -p project1 -p project2 $VERSION

# Associate commits with the release
sentry-cli releases set-commits --auto $VERSION
```

> If you do not have a repo-based integration associated with your sentry organization, then the `--auto` flag will automatically use the git tree of your local repo, and associate commits between the previous release’s commit and the current head commit with the release. If this is the first release, we’ll use the latest 20 commits. This behaviour is configurable with the `--initial-depth` flag. 

Reviewer: ```suggestion
If you don't have a repo-based integration associated with your sentry organization, then the `--auto` flag will automatically use the git tree of your local repo, and associate commits between the previous release's commit and the current head commit with the release. If this is the first release, Sentry will use the latest 20 commits. This behavior is configurable with the `--initial-depth` flag. 
```

Author: lol the Canadian version `behaviour` 😄

> Alternatively, you can use the the `--local` flag to enable this behaviour by default.

Reviewer: ```suggestion
Alternatively, you can use the `--local` flag to enable this behavior by default.
```
